jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/go/GoHelper.kt

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.go
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectFileIndex
+import com.intellij.openapi.vfs.VirtualFile
+
+/**
+ * "Light" ides like Goland do not rely on marking folders as source root, so infer it based on
+ * the go.mod file
+ *
+ * @throws IllegalStateException If the contentRoot cannot be located
+ */
+fun inferSourceRoot(project: Project, virtualFile: VirtualFile): VirtualFile? {
+    val projectFileIndex = ProjectFileIndex.getInstance(project)
+    return projectFileIndex.getContentRootForFile(virtualFile)?.let {
+        findChildGoMod(virtualFile.parent, it)
+    }
```

Comment:
Should this just do a find up? instead of up and back down?

Reply:
Fixed